handler/src/main/java/io/netty/handler/ssl/OpenSslX509KeyManagerFactory.java

```diff
@@ -218,4 +230,147 @@ void destroy() {
+
+                @Override
+                public void engineSetKeyEntry(String alias, Key key, char[] password, Certificate[] chain)
+                        throws KeyStoreException {
+                    throw new KeyStoreException("Not supported");
+                }
+
+                @Override
+                public void engineSetKeyEntry(String alias, byte[] key, Certificate[] chain) throws KeyStoreException {
+                    throw new KeyStoreException("Not supported");
+                }
+
+                @Override
+                public void engineSetCertificateEntry(String alias, Certificate cert) throws KeyStoreException {
+                    throw new KeyStoreException("Not supported");
+                }
+
+                @Override
+                public void engineDeleteEntry(String alias) throws KeyStoreException {
+                    throw new KeyStoreException("Not supported");
+                }
+
+                @Override
+                public Enumeration<String> engineAliases() {
+                    return Collections.enumeration(Collections.singleton(SslContext.ALIAS));
+                }
+
+                @Override
+                public boolean engineContainsAlias(String alias) {
+                    return SslContext.ALIAS.equals(alias);
+                }
+
+                @Override
+                public int engineSize() {
+                    return 1;
+                }
+
+                @Override
+                public boolean engineIsKeyEntry(String alias) {
+                    return engineContainsAlias(alias);
+                }
+
+                @Override
+                public boolean engineIsCertificateEntry(String alias) {
+                    return engineContainsAlias(alias);
+                }
+
+                @Override
+                public String engineGetCertificateAlias(Certificate cert) {
+                    if (cert instanceof X509Certificate) {
+                        for (X509Certificate x509Certificate : certificateChain) {
+                            if (x509Certificate.equals(cert)) {
+                                return SslContext.ALIAS;
+                            }
+                        }
+                    }
+                    return null;
+                }
+
+                @Override
+                public void engineStore(OutputStream stream, char[] password)
+                        throws IOException, NoSuchAlgorithmException, CertificateException {
+                    throw new UnsupportedOperationException();
+                }
+
+                @Override
+                public void engineLoad(InputStream stream, char[] password)
+                        throws IOException, NoSuchAlgorithmException, CertificateException {
+                    if (stream != null && password != null) {
+                        throw new UnsupportedOperationException();
+                    }
+                }
+            }, null, "native");
```

Comment:
`"native"` is also used in `OpenSslPrivateKey#getFormat` ... should we share a constant here?

Reply:
this was actually a bug... We should return `null` from `OpenSslPrivateKey#getFormat` as we do not support encoding.